### PR TITLE
Test, CI, and CMake

### PR DESCRIPTION
**Pull request by kgerheiser**
@aerorahul I just got rid of shared libraries. Most NCEPLIBS doesn't even build with them because of some dependencies.

I also removed the version in the library name

### CODE REVIEW

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -1,116 +1,88 @@
...
+  set(CMAKE_C_FLAGS_RELEASE "-O3")
+  set(CMAKE_Fortran_FLAGS "-ggdb -fno-range-check -fbacktrace -funroll-loops")
+  set(CMAKE_Fortran_FLAGS_RELEASE "-O3")
+  set(fortran_d_flags "-fdefault-real-8")
+  set(fortran_8_flags "-fdefault-integer-8 -fdefault-real-8")
+endif()
+
+set_source_files_properties(${c_src} PROPERTIES COMPILE_OPTIONS "${c_flags}")
+set_source_files_properties(${c_src} PROPERTIES COMPILE_DEFINITIONS
+  "${platform_definitions}")
```
Comment (aerorahul) [inline src/CMakeLists.txt]: variable `platform_definitions` is not defined and hence this line does nothing.
please remove.

**Review by BoiVuong-NOAA — APPROVED** [review]
Look good !

File: src/list_of_files.cmake [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+  ${CMAKE_CURRENT_SOURCE_DIR}/w3pradat.f
+  ${CMAKE_CURRENT_SOURCE_DIR}/w3reddat.f
+  ${CMAKE_CURRENT_SOURCE_DIR}/w3tagb.f
+  ${CMAKE_CURRENT_SOURCE_DIR}/w3trnarg.f
+  ${CMAKE_CURRENT_SOURCE_DIR}/w3unpk77.f
+  ${CMAKE_CURRENT_SOURCE_DIR}/w3utcdat.f
+  ${CMAKE_CURRENT_SOURCE_DIR}/w3valdat.f
+  ${CMAKE_CURRENT_SOURCE_DIR}/w3ymdh4.f
+  ${CMAKE_CURRENT_SOURCE_DIR}/xmovex.f
+  ${CMAKE_CURRENT_SOURCE_DIR}/xstore.f
```
Comment (aerorahul) [inline src/list_of_files.cmake]: `list_of_files.cmake` is in the same directory as `src`. `CMAKE_CURRENT_SOURCE_DIR` is not necessary.
Please remove.

File: src/list_of_files.cmake [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+  )
+
+set(c_src
+  ${CMAKE_CURRENT_SOURCE_DIR}/mova2i.c
+  ${CMAKE_CURRENT_SOURCE_DIR}/summary.c
```
Comment (aerorahul) [inline src/list_of_files.cmake]: same comment as above for `fortran_src` files.

File: cmake/PackageConfig.cmake.in [hunk 4]
```diff
@@ -0,0 +1,18 @@
+@PACKAGE_INIT@
```
Comment (aerorahul) [inline cmake/PackageConfig.cmake.in:1]: I am glad you removed the shared library targets.

**Review by aerorahul — COMMENTED** [review]
looks good. minor comments inline.

**Pull request merged**